API.md

```diff
@@ -0,0 +1,136 @@
+# Mail Ballot Manager APIs
+
+## Open Questions
+
+- If there is an error in the election data, how does a election official start
+  over?
+- If there is an error in the voter mailing list, how does an election official
+  fix it?
+- If an election official needs to stop the presses, can they?
+
+## Types
+
+Election as defined in `@votingworks/ballot-encoder` and the following…
+
+```
+interface VoterMailBallot {
+  id: string,
+  firstName: string,
+  middleName: string,
+  lastName: string,
+  nameSuffix: string,
+  street1: string,
+  street2: string,
+  city: string,
+  state: string,
+  zipcode: string,
+  ballotStyle: string,
+  precinctId: string,
+  ballotfile: string, // filepath?
+  ballotCreated: timestamp,
+  ballotPrinterReceived: timestamp,
+  ballotPrinterPrinted: timestamp
+  ballotOutboundSent: timestamp
+  ballotOutboundDelivered: timestamp
+  ballotInboundSent: timestamp
+  ballotInboundDelivered: timestamp
+}
+```
+
+```
+interface BallotTemplate {
+  fileName: string,
+  fileContent: pdfFile,     // or filepath? Perhaps different for upload vs list.
+  ballotStyle: BallotStyle, // As as defined in `@votingworks/ballot-encoder`
+  precinct: Precinct,       // As as defined in `@votingworks/ballot-encoder`
+}
+```
+
+```
+interface InsertsData {
+  affadavit: string,
+  instructions: string,
+  helpPhone: string,
+  helpEmail: string,
+  helpWeb: string,
+  helpAddress: string,
+}
+```
+
+## Auth API
+
+Whatever Auth0 uses here.
+
+## Users API
+
+What user date is provided by default via Auth0?
+
+### `GET /auth/me`
+
+```
+{
+  id: string,
+  email: string,
+}
+```
+
+## Elections API
+
+### `GET /elections/list`
+
+```
+[{
+  id: string,
+  definition: Election,
+  packageHash: string,
+  printingAndMailingApprovedAt: datetime,
```

Comment:
I'm assuming `datetime` is a ISO-8601 formatted string?

Reply:
Yes, ideally.